projecttacoma/bulk-submit-demo
# Initial Setup of Mantine Components and add Measure ID and ExportURL inputs

## Summary
Adds basic [Mantine](https://mantine.dev/) setup to the project with basic header bar and light/dark mode switch. Also the start of the input components Measure selection and ExportURL input. Additional sets up Jest unit testing of components.

### New Behavior
No longer default Next.JS page. Runs on port 3002 in dev mode. Loads measure ids from `deqm-test-server` to populate measure id selector. Shows error notifications if there were issues with getting measure listing to load.

### Code Changes
Added `@mantine/core`, `@mantine/next`, `@mantine/hooks`, `@mantine/notifications` dependencies for Mantine setup.
Added [`tabler-icons-react`](https://tabler-icons-react.vercel.app/) dependency for MIT licensed icons that work well with Mantine.
Added `@types/fhir` dev dependency to help with TS based FHIR JSON.
Created `components` directory for our custom components and created components for our app header, measure input field and export url field.
Created and updated `_app.tsx`, `_document.tsx`, `index.tsx` to support Mantine use in next and make use of our custom components.
Configured Jest unit tests located in `__tests__` with `testHelpers.tsx` file to help with wrapping the tested components with necessary Mantine providers.

## Testing Guidance
- Start `deqm-test-server` loaded with measures.
- Start demo app in dev mode with `npm run dev`.
- Navigate to `http://localhost:3002`, make sure Measure listing loads.
- Kill `deqm-test-server`.
- Reload `http://localhost:3002`, make sure you see error(s) indicating that measure listing could not be loaded.
